### PR TITLE
Correctly report single-step agent installations on Docker

### PR DESCRIPTION
**Pull request by yshapiro-57**
### Motivation

TL;DR: This PR will let us collect more accurate telemetry on how often customers use single-step APM instrumentation for agent installs on Docker.

This PR addresses an issue from https://github.com/DataDog/agent-linux-install-script/pull/131 : even though the code in that PR sets the install type to `linux_single_step_dkr` for Docker agent installations, we are not actually seeing any of these values on our backend, because for Docker, the install script does not run in the same container as the agent. So any install signature that the install script might write is not going to be visible to the agent anyway. In fact, the install script does not even attempt to write the signature for Docker installs, because then [no_agent is set to true](https://github.com/DataDog/agent-linux-install-script/blob/f82603d43f9b805fff781471f16566d6a0e9e4a6/install_script.sh.template#L707) and as a result [we make no attempt to write an install signature](https://github.com/DataDog/agent-linux-install-script/blob/f82603d43f9b805fff781471f16566d6a0e9e4a6/install_script.sh.template#L1743-L1747)

After discussing with the Container Integrations team, passing special environment variables (install ID, install type and install time) from the install script into the dockerized agent is not trivial; they aren't aware of an easy way to do it. Instead in this PR I am making the agent report one of two new install types, `docker_manual` or `docker_single_step`, if it realizes from looking at the environment that it is running inside a Docker container. If the `DOCKER_DD_AGENT` env variable is set, then we are on docker. If the `DD_APM_ENABLED` environment variable is also set, then this is a single step Docker agent installation. Otherwise we report a manually instrumented Docker agent installation.
`

### Additional Notes

Jira: [Set install ID, time and type for Docker](https://datadoghq.atlassian.net/browse/AIT-10210)

### Possible Drawbacks / Trade-offs

It is theoretically possible for a customer to mess with the environment variables in their Docker container, but I am hopeful that only a small minority of customers will.

### Describe how to test/QA your changes

Make sure all new unit tests pass.

Deploy an agent image from CI for this build on Docker, and make sure the install signature is set correctly, as follows:

- There is a CI job on this PR called `dd-gitlab/docker_build_agent7_arm64`, like [this one](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/507531474). Open the link to that job and copy the name of the Docker image that it built: `486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent/agent:v33896657-c318ad95-7-arm64` .
- Download that Docker image locally:
```
aws-vault exec sso-build-stable-developer -- docker pull 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent/agent:v33896657-c318ad95-7-arm64
```
- Start a Docker container using that downloaded Docker image:
```
 docker run -d --name dd-agent \
-e DD_API_KEY=your_api_key \
-e DD_SITE="datadoghq.com" \
-e DD_APM_ENABLED=true \
-e DD_APM_NON_LOCAL_TRAFFIC=true \
-e DD_APM_RECEIVER_SOCKET=/opt/datadog/apm/inject/run/apm.socket \
-e DD_DOGSTATSD_SOCKET=/opt/datadog/apm/inject/run/dsd.socket \
-v /opt/datadog/apm:/opt/datadog/apm \
-v /var/run/docker.sock:/var/run/docker.sock:ro \
-v /proc/:/host/proc/:ro \
-v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
-v /var/lib/docker/containers:/var/lib/docker/containers:ro \
486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent/agent:v33896657-c318ad95-7-arm64
```
- Exec into that container that you have just started:
```
docker exec -i -t dd-agent bash
```
Check the agent install signature:
```
root@2c6f4a3bbc5b:/# cat /etc/datadog-agent/install.json
{"install_id":"e029d97a-2223-4f35-b0a2-b6b5734f48bd","install_type":"docker_single_step","install_time":1715200376}
root@2c6f4a3bbc5b:/#
```
This confirms that the agent install signature for a single-step Agent installation on Docker now correctly sets the install type to `docker_single_step` instead of `manual`.

Stop and delete your container through the Docker UI, and then start it with the same command, but without `-e DD_APM_ENABLED=true`. Verify that the install type on the container is now set to `docker_manual`:
```
root@17005dd3d8d3:/# cat /etc/datadog-agent/install.json
{"install_id":"7db4258b-e63e-479a-ba9f-ae57a9e4a20f","install_type":"docker_manual","install_time":1715206186}
root@17005dd3d8d3:/#
```

### CODE REVIEW

File: comp/trace/config/install_signature_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInferInstallTypeFromEnvironment(t *testing.T) {
+	t.Run("Non-dockerized agent", func(t *testing.T) {
+		assert.Equal(t, defaultInstallType, inferInstallTypeFromEnvironment())
+	})
+
+	t.Run("Dockerized manual instrumentation", func(t *testing.T) {
+		_ = os.Setenv("DOCKER_DD_AGENT", "true")
```
[the remark this answers is not in the document]
Comment (yshapiro-57) [reply comp/trace/config/install_signature_test.go]: @ajgajg1134 thank you! I've just changed it to `t.Setenv`.

**Review by ajgajg1134 — APPROVED** [review]
looks good from agent-apm!
